File: analytics/src/main/java/io/appmetrica/analytics/impl/component/clients/MainCommutationClientUnitFactory.java
package io.appmetrica.analytics.impl.component.clients;

import android.content.Context;
import androidx.annotation.NonNull;
import io.appmetrica.analytics.impl.component.CommonArguments;
import io.appmetrica.analytics.impl.component.CommutationComponentId;
import io.appmetrica.analytics.impl.component.CommutationDispatcherComponent;
import io.appmetrica.analytics.impl.component.CommutationDispatcherComponentFactory;

//For [Idle/Anonymous/Commutation]ReporterEnvironment
public class MainCommutationClientUnitFactory implements ClientUnitFactory {

    @NonNull
    public CommutationClientUnit createClientUnit(@NonNull Context context,
                                                  @NonNull ComponentsRepository repository,
                                                  @NonNull ClientDescription clientDescription,
                                                  @NonNull CommonArguments sdkConfig) {

        CommutationDispatcherComponent componentUnit = repository.getOrCreateCommutationComponent(
                new CommutationComponentId(clientDescription.getPackageName()),
                sdkConfig,
                new CommutationDispatcherComponentFactory()
        );

        return new CommutationClientUnit(
                context,
                componentUnit,
                sdkConfig
        );
    }
}

File: analytics/src/main/java/io/appmetrica/analytics/impl/component/clients/MainCommutationClientUnitFactory.kt
package io.appmetrica.analytics.impl.component.clients

import android.content.Context
import io.appmetrica.analytics.impl.component.CommonArguments
import io.appmetrica.analytics.impl.component.CommutationComponentId
import io.appmetrica.analytics.impl.component.CommutationDispatcherComponentFactory

// For [Idle/Anonymous/Commutation]ReporterEnvironment
internal class MainCommutationClientUnitFactory : ClientUnitFactory<CommutationClientUnit> {
    override fun createClientUnit(
        context: Context,
        repository: ComponentsRepository,
        clientDescription: ClientDescription,
        sdkConfig: CommonArguments
    ): CommutationClientUnit {
        val componentUnit = repository.getOrCreateCommutationComponent(
            CommutationComponentId(clientDescription.packageName),
            sdkConfig,
            CommutationDispatcherComponentFactory()
        )

        return CommutationClientUnit(
            context,
            componentUnit,
            sdkConfig
        )
    }
}
